app/controllers/activities_controller.rb

```diff
@@ -28,7 +28,7 @@ def set_activities
   end
 
   def teams
-    Team.in_current_season.accepted.order(:name)
+    (Team.in_current_season.accepted.presence || Team.in_previous_season.accepted).order(:name)
```

Comment:
Cool!
Do you think we could make this even simpler as `Team.accepted.in_nearest_season` , where `in_nearest_season` is a new scope for in current season OR in previous season?
(Note, we do have an index teams_on_season, not on teams_on_kind. Maybe worth to add that if you want to go with ^)

Reply:
@F3PiX I tried adding the index on `kind` and was trying to use ActiveRecord's `explain` to see if the index was being used or not. I couldn't get it to be used on `Team.accepted`, only when filtering specific kinds.

<img width="1081" alt="screen shot 2018-06-03 at 00 46 22" src="https://user-images.githubusercontent.com/31735/40881792-51b84376-66c8-11e8-9e81-059bc0623ddd.png">

<img width="718" alt="screen shot 2018-06-03 at 00 46 47" src="https://user-images.githubusercontent.com/31735/40881794-55b71e66-66c8-11e8-8f7b-42d82e2bee1c.png">

(I had to create 3000 teams for the index to be used, by the way. Postgres was not using it even with 200 teams.)

Here's `Team.in_nearest_season.accepted`:

Without `kind` index:

<img width="981" alt="screen shot 2018-06-03 at 00 55 22" src="https://user-images.githubusercontent.com/31735/40881818-d99d79aa-66c8-11e8-953e-a4859306a6f6.png">

With the `kind` index:

<img width="981" alt="screen shot 2018-06-03 at 00 53 54" src="https://user-images.githubusercontent.com/31735/40881821-e158b6aa-66c8-11e8-87e5-fcbab9098f13.png">

I only had 2 seasons for this experiment - not sure if that matters too much (I don't think so).

So, I don't know if it's worth it. But I can commit it if you think it makes sense! 👍

Reply:
> I couldn't get it to be used on Team.accepted, only when filtering specific kinds.

Oooo. I didn't think of that. Is that because `accepted` is not a field in the database? I don't know enough about this, but maybe a compound index on season and kind would work with accepted? Or is that the same difference? I really don't know. Interesting!

> So, I don't know if it's worth it.  

Well, if it doesn't use the index with the `accepted` scope.... no. 
I didn't expect a lot better on this particular query, but we mostly use `teams` as accepted teams, so I'd expect a better performance over all. 
But again, not if the index is not being used.

Anyway, no reason to keep this PR on hold over this.  
Thank you so much for the clarification!

Reply:
> Is that because `accepted` is not a field in the database?

I think it's because the SQL query is an includes query:

```SQL
SELECT "teams".* FROM "teams" WHERE ("teams"."kind" IN ('full_time', 'sponsored') OR "teams"."kind" IN ('part_time', 'voluntary'))
```

I tried the query without the `OR` and it also did not use the index.

<img width="904" alt="screen shot 2018-06-03 at 13 43 03" src="https://user-images.githubusercontent.com/31735/40886655-1a386d42-6734-11e8-93d2-785e261a3132.png">

I think it may be because the result of this query in my local database is the entire teams table - apparently I don't have companies that are not accepted 😛 Yep, I think that's why. Here's me trying an includes query with values that don't exist:

<img width="631" alt="screen shot 2018-06-03 at 13 49 42" src="https://user-images.githubusercontent.com/31735/40886877-9d5b0fd8-6737-11e8-99d0-72ab3c55536f.png">

As you can see it uses the index. Here's more context I found [here](https://robots.thoughtbot.com/why-postgres-wont-always-use-an-index):

> Given a large enough number of requested rows (such as all the posts with an id greater than 10), it is actually more efficient to go fetch all those rows from the table directly rather than to go through the extra step of asking the index for the position of those rows and next fetch those specific rows from the table.
> If the percentage of the rows is smaller than 5-10% of all rows in the table the benefit of using the information stored in the index outweighs the additional intermediate step.

I don't know how our production data look like, but I assume the vast majority of our teams are accepted, so it's likely the index will never be used for the `accepted` scope. But I'm no expert! It was fun exploring this though 😸 Thanks.

Didn't see changes with a compound kind, season_id index. Didn't explore much further though.

Reply:
Very interesting stuff! Thanks for the background info! BTW, based on that info and as not all applying teams are accepted, I think the case for an index is getting stronger!

Reply:
Yep, given only 7.5% of teams in production are accepted, I think this makes sense 🚀

Reply:
Yep, given only a fraction of teams in production are accepted, I think this makes sense 🚀